File: bs.py
import numpy as np
import sys

def f(x):
    n = 130869380
    t = n / 4
    d = 1.41e-10
    y = 1 + n * np.exp(-d * (n + 1) * x)
    return t - (n + 1) / y


a, b = map(float, input().split(" "))

delta = 3.61034e-7

if f(a) * f(b) >= 0:
    print("bosta")
    sys.exit(0)

w = 0
if f(a) > f(b):
    w = 1

cnt = 0
def bs(l, r):
    mid = (l + r) / 2.0
    global cnt
    cnt += 1
    print("iter:", cnt, "%.15f" % mid, "%.15f" % f(mid))
    if cnt == 20:
        return mid
    if f(mid) < 0:
        if w == 1:
            return bs(l, mid)
        else:
            return bs(mid , r)
    if w == 1:
        return bs(mid, r)
    else:
        return bs(l, mid)

print(bs(a, b))

File: bs.cpp
#include <bits/stdc++.h>

using namespace std;

#ifndef ONLINE_JUDGE
#include "debug.h"
#else
#define dbg(x...)
#endif

// busca binaria padrao
int main(){
  ios_base::sync_with_stdio(false);
  cin.tie(nullptr);
  auto f = [&](long double x) {
      return 3040754 - 1241839 * exp(x) - (exp(x) - 1) * 408181.0 / x;
  };
  long double a,b;
  cin >> a >> b;
  dbg(a,b);
  if (f(a) * f(b) >= 0) {
    cout << "merda\n";
    return 0;
  }
  const long double delta = 3.61034e-7;
  int cnt = 0;
  cout << setprecision(15) << fixed;
  bool w = f(a) > f(b);
  function<long double(long double,long double)> bs = [&](long double l, long double r) {
    long double mid = (l + r) / 2.0;
    cout << "iter: " << (++cnt) << ' ';
    cout << mid << ' ' << f(mid) << '\n';
    if (abs(f(mid)) < delta) return f(mid);
    if (f(mid) < 0) { 
      return w ? bs(l, mid) : bs(mid, r);
    }
    return w ? bs(mid, r) : bs(l, mid);
  };
  cout << bs(a,b) << '\n';
}





